tests/ops/test_hamiltonian.py

```diff
@@ -404,6 +404,103 @@ def test_hamiltonian_matmul(self):
         assert H.compare(H1 @ H2)
 
 
+class TestGrouping:
+    """Tests for the grouping functionality"""
+
+    def test_grouping_is_correct(self):
+        """Basic test checking that grouping works as expected"""
+        a = qml.PauliX(0)
+        b = qml.PauliX(1)
+        c = qml.PauliZ(0)
+        obs = [a, b, c]
+        coeffs = [1.0, 2.0, 3.0]
+
+        H = qml.Hamiltonian(coeffs, obs)
+        assert H.grouping_indices is None
+
+        # compute grouping during `get_grouping`
+        grouped_coeffs, grouped_obs = H.get_grouping()
+        assert np.allclose(grouped_coeffs[0], np.array([1.0, 2.0]))
+        assert np.allclose(grouped_coeffs[1], np.array(3.0))
+        assert grouped_obs == [[a, b], [c]]
+        assert H.grouping_indices == [[0, 1], [2]]
+
+        # compute grouping during construction
```

Comment:
This is just personal preference but maybe split this test into three so that each of the different construction methods are tested separately?